core/frontend/src/components/logs/LogManager.vue

```diff
@@ -0,0 +1,231 @@
+        </v-btn>
+        <v-btn
+          icon
+          @click="removeLogs"
+        >
+          <v-icon>mdi-trash-can</v-icon>
+        </v-btn>
+      </v-card-title>
+      <v-card-text>
+        <v-data-table
+          v-model="selected_logs"
+          :headers="headers"
+          :items="parsed_logs"
+          item-key="name"
+          show-select
+        >
+          <template #item.actions="{ item }">
+            <v-icon @click="replay_log(item)">
+              mdi-play
+            </v-icon>
+            <v-icon @click="downloadLogs([item])">
+              mdi-download
+            </v-icon>
+          </template>
+        </v-data-table>
+      </v-card-text>
+    </v-card>
+    <v-container v-else>
+      <spinning-logo />
+    </v-container>
+  </v-container>
+</template>
+
+<script lang="ts">
+import axios, { AxiosPromise } from 'axios'
+import { format } from 'date-fns'
+import Vue from 'vue'
+import { getModule } from 'vuex-module-decorators'
+
+import NotificationsStore from '@/store/notifications'
+import { log_service } from '@/types/frontend_services'
+import { LiveNotification, NotificationLevel } from '@/types/notifications'
+
+import SpinningLogo from '../common/SpinningLogo.vue'
+
+const notification_store: NotificationsStore = getModule(NotificationsStore)
+
+interface FileBrowserFile {
+  path: string
+  name: string
+  size: number
+  extension: string
+  modified: string
+  mode: number
+  isDir: boolean
+  type: string
+}
+
+const log_folders = [
+  { name: 'Autopilot logs', path: '/root/.config/ardupilot-manager/firmware/logs/' },
+  { name: 'Telemetry logs', path: '/root/.config/ardupilot-manager/logs/' },
+]
+
+const baseURL = '/file-browser/api'
+
+export default Vue.extend({
+  name: 'LogManager',
+  components: {
+    SpinningLogo,
+  },
+  data() {
+    return {
+      auth_key: '',
+      available_logs: [] as FileBrowserFile[],
+      fetching_logs: false,
+      selected_logs: [] as FileBrowserFile[],
+      headers: [
+        {
+          text: 'Name',
+          align: 'start',
+          sortable: false,
+          value: 'name',
+        },
+        { text: 'Size (MB)', value: 'size' },
+        { text: 'Type', value: 'extension' },
+        { text: 'Modified', value: 'modified' },
+        {
+          text: 'Actions',
+          align: 'end',
+          sortable: false,
+          value: 'actions',
+        },
+      ],
+    }
+  },
+  computed: {
+    parsed_logs(): FileBrowserFile[] {
+      return this.available_logs.map((log) => ({
+        ...log,
+        modified: format(new Date(log.modified), 'yyyy-MM-dd HH:mm:ss'),
+        size: log.size / 1e6,
+      }))
+    },
+    are_logs_available(): boolean {
+      return this.available_logs.length !== 0
+    },
+  },
+  async mounted() {
+    await this.authenticateOnFileBrowser()
+    await this.fetchAvailableLogs()
+  },
+  methods: {
+    async authenticateOnFileBrowser(): Promise<void> {
```

Comment:
We should break all this file handle logic API in a different class to be common for all our components.

Reply:
Done!